juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+[comment]: <> (How To Perform Fixing Attach Volume to Instance)
+
+## pre-install
+
+Duration: 120:00
+
+This description of how the deployment of `charmed kubernetes` is performed on a OpenStack server, where OpenStack is deployed to another OpenStack. (openstack-on-openstack).
+
+There are two assumptions:
+1. juju is configured to use the OpenStack cloud
+2. openstack is configured with valid credentials (e.g. `source novarc`)
+
+### OpenStack
+
+Deploy the openstack-base bundle with the custom overlay.
+
+##### Memory overlay
+```yaml
+applications:
+  nova-compute:
+    annotations:
+    constraints: mem=8G
+```
+
+##### Deployment
+
+```bash
+git clone https://github.com/openstack-charmers/openstack-bundles.git
+juju deploy ./openstack-bundles/stable/openstack-base/bundle.yaml --overlay ./openstack-bundles/stable/overlays/openstack-base-virt-overlay.yaml --overlay ./memory-overlay.yaml
+```
+
+##### initialize and unseal vault and create self-signed root CA
+
+1. Vault need to be installed.
+   ```bash
+   snap install vault
+   ```
+2. Wait for the `vault/0` unit to be in the `blocked` state with message `Vault needs to be initialized`.
+3. Initialize the vault deployment.
+   ```bash
+   export VAULT_ADDR="http://$(juju status | grep vault/0 | tr -s " " | cut -d" " -f5):8200"
+   vault operator init -key-shares=5 -key-threshold=3 > vault-keys
```

Comment:
Would it be possible to remove the "1." item and use `juju ssh vault/0` and then `export VAULT_ADDR="http://127.0.0.1:8200"` instead? The vault unit will already have the vault snap installed so no need of manually installing the snap anywhere else.

Reply:
You're right, I changed it to run on the `vault/0` unit.